src/codegen/cfg.rs

```diff
@@ -678,7 +680,14 @@ impl ControlFlowGraph {
                 self.expr_to_string(contract, ns, l),
                 self.expr_to_string(contract, ns, r)
             ),
-            Expression::Variable(_, _, res) => format!("%{}", self.vars[res].id.name),
+            Expression::Variable(_, _, res) => {
+                if let Some(var) = self.vars.get(res) {
+                    format!("%{}", var.id.name)
+                } else {
+                    eprintln!("error: non-existing variable {} in CFG", res);
```

Comment:
This only prints the error but doesn't do anything about it? I'd guess that this result in invalid code or causes problems later on, as the old code would panic instead?

Reply:
I've changed it to a panic for now.